modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TagsRatingTestCase.java

```diff
@@ -0,0 +1,131 @@
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init(userMode);
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Comment Rating Test case")
+    public void testTagsTestCase() throws Exception {
+        String APIName = "CommentRatingAPI";
+        String APIContext = "commentRating";
+        String tags = "youtube, video, media";
+        String url = getGatewayURLHttp() + "jaxrs_basic/services/customers/customerservice";
+        String description = "This is test API create by API manager integration test";
+        String providerName = user.getUserName();
+        String APIVersion = "1.0.0";
+        String apiData;
+
+        APIRequest apiRequest = new APIRequest(APIName, APIContext, new URL(url));
+        apiRequest.setTags(tags);
+        apiRequest.setDescription(description);
+        apiRequest.setVersion(APIVersion);
+        apiRequest.setProvider(providerName);
+
+        //add api
+        HttpResponse serviceResponse = restAPIPublisher.addAPI(apiRequest);
+        apiId = serviceResponse.getData();
+
+        //publish the api
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, APILifeCycleAction.PUBLISH.getAction(), null);
+
+        //get api
+        HttpResponse serviceGetResponse = restAPIPublisher.getAPI(apiId);
+        apiData = serviceGetResponse.getData();
+        JSONObject apiObject = new JSONObject(apiData);
+
+        // Test tags
+        JSONArray tagsList = (JSONArray) apiObject.get("tags");
+
+        for (int i = 0; i < tagsList.length(); i++) {
+            assertTrue(tags.contains(tagsList.getString(i)), "API tag data mismatched");
+        }
+    }
+
+    @Test(dependsOnMethods = {"testTagsTestCase"}, groups = { "wso2.am" }, description = "Test and Verify Added Rating Test Case")
+    public void testRatingTest() throws Exception {
+
+        //Test and verify added rating
+        Integer rating = 4;
+        HttpResponse ratingAddResponse = restAPIStore
+                .addRating(apiId, 4, gatewayContextWrk.getContextTenant().getDomain());
+        assertEquals(ratingAddResponse.getResponseCode(), Response.Status.OK.getStatusCode(), "Error adding rating");
+        Gson getRatingsGson = new Gson();
+        RatingDTO ratingDTO = getRatingsGson.fromJson(ratingAddResponse.getData(), RatingDTO.class);
+        assertEquals(ratingDTO.getRating(), rating, "Ratings do not match");
+
+        //Test delete rating
+        HttpResponse deleteRatingResponse = restAPIStore
+                .removeRating(apiId, gatewayContextWrk.getContextTenant().getDomain());
+        assertEquals(deleteRatingResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+                "Response code mismatched");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        restAPIPublisher.deleteAPI(apiId);
+        super.cleanUp();
+    }
+}
```

Comment:
Add new line at the end of file.

Reply:
Fixed with https://github.com/wso2/product-apim/pull/11163/commits/909e61ea2edc8a1e3f88aead2a5213324d94bd84